spec/controllers/expenses_controller_spec.rb

```diff
@@ -0,0 +1,132 @@
+require "rails_helper"
+
+describe ExpensesController do
+  context "GET #new" do
+    context "when invalid params" do
+      it "won't create a new expense" do
+        user = create(:user)
+        create(:budget, user: user)
+        allow(controller).to receive(:current_user).and_return(user)
+        expense_params = { name: "Gas", bill_estimate: 500, frequency: "" }
+
+        expect do
+          post :create, params: { expense: expense_params }
+        end.to change(Expense, :count).by(0)
+      end
+
+      it "renders #new" do
+        user = create(:user)
```

Comment:
I think this spec file is a good example for when we call pull out a helper method.
A suggestion to add at the bottom of this file:
```
def create_current_user_with_budget
  user = create(:user)
  create(:budget, user: user)
  allow(controller).to receive(:current_user).and_return(user)
end
```
And then use it every spec that uses this structure

Reply:
Great idea! Done in sha: e427e21